@xen-orchestra/rest-api/src/users/user.controller.mts

```diff
@@ -0,0 +1,59 @@
+import { Example, Get, Path, Query, Request, Response, Route, Security, Tags } from 'tsoa'
+import type { Request as ExRequest } from 'express'
+import { provide } from 'inversify-binding-decorators'
+import { notFoundResp, unauthorizedResp, type Unbrand } from '../open-api/common/response.common.mjs'
+import { XoController } from '../abstract-classes/xo-controller.mjs'
+import type { WithHref } from '../helpers/helper.type.mjs'
+import type { XoUser } from '@vates/types'
+import { partialUsers, user, userIds } from '../open-api/oa-examples/user.oa-example.mjs'
+
+@Route('users')
+@Security('*')
+@Response(unauthorizedResp.status, unauthorizedResp.description)
+@Tags('users')
+@provide(UserController)
+export class UserController extends XoController<XoUser> {
+  // --- abstract methods
+  getAllCollectionObjects(): Promise<XoUser[]> {
+    return this.restApi.xoApp.getAllUsers()
+  }
+  getCollectionObject(id: XoUser['id']): Promise<XoUser> {
+    return this.restApi.xoApp.getUser(id)
+  }
```

Comment:
IMO, these methods should return sanitized users. This way, every time we call `this.getObject/this.getObjects`, we will get a user without a password. The REST API does not need to manage user passwords.